cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/statements/DoStatement.kt

```diff
@@ -50,6 +51,15 @@ class DoStatement : Statement() {
             .append("statement", statement)
             .toString()
 
+    override fun addArgument(expression: Expression) {
+        this.condition = expression
+    }
+
+    override fun replaceArgument(old: Expression, new: Expression): Boolean {
+        this.condition = new
```

Comment:
This should check whether this condition holds "old" or not and only replace if condition is "old" and return true/false accordingly.